demisto_sdk/commands/format/update_mapper.py

```diff
@@ -55,3 +61,41 @@ def set_mapping(self):
         """
         if not self.data.get('mapping'):
             self.data['mapping'] = {}
+
+    def extract_content_fields(self, content_fields, built_in_fields):
+        """
+        Extract fields which only exist in the id set file.
+        """
+        def _extract_content_fields(field):
+            inc_name, inc_info = field
+            # incoming mapper
+            if self.data.get('type', {}) == "mapping-incoming":
+                if inc_name in content_fields or inc_name.lower() in built_in_fields:
+                    return True
+            # outgoing mapper
+            if self.data.get('type', {}) == "mapping-outgoing":
+                # for inc timer type: "field.StartDate, and for using filters: "simple": "".
+                if simple := inc_info.get('simple'):
+                    if '.' in simple:
+                        simple = simple.split('.')[0]
+                    if simple in content_fields or simple in built_in_fields:
+                        return True
+            return False
+
+        return _extract_content_fields
+
+    def remove_inexistent_fields(self):
+        """
+        Remove in-existent fields from a mapper.
+        """
+        content_fields = get_all_incident_and_indicator_fields_from_id_set(self.id_set_file, 'mapper')
+        built_in_fields = [field.lower() for field in BUILT_IN_FIELDS] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
+
+        mapper = self.data.get('mapping', {})
+        for mapping_name in mapper.values():
+            mapping_name['internalMapping'] = dict(
+                filter(
```

Comment:
I think a dictionary comprehension looks more pythonic
`{k : v for (k,v) in mapping_name.get('internalMapping', {}).items() if self.extract_content_fields(content_fields=content_fields, built_in_fields=built_in_fields) }` 
(lets use this formula for the other filters as well)

Reply:
self.extract_content_fields should be replaced here with the new predicate